nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-restapi-processors/src/test/java/org/apache/nifi/processors/elasticsearch/AbstractPaginatedJsonQueryElasticsearchTest.java

```diff
@@ -0,0 +1,350 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License") you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.elasticsearch;
+
+import org.apache.nifi.processors.elasticsearch.api.AggregationResultsFormat;
+import org.apache.nifi.processors.elasticsearch.api.PaginationType;
+import org.apache.nifi.processors.elasticsearch.api.ResultOutputStrategy;
+import org.apache.nifi.processors.elasticsearch.api.SearchResultsFormat;
+import org.apache.nifi.provenance.ProvenanceEventType;
+import org.apache.nifi.util.MockFlowFile;
+import org.apache.nifi.util.TestRunner;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public abstract class AbstractPaginatedJsonQueryElasticsearchTest extends AbstractJsonQueryElasticsearchTest<AbstractPaginatedJsonQueryElasticsearch> {
+   private static final String TEST_DIR = "src/test/resources/AbstractPaginatedJsonQueryElasticsearchTest";
+    private static final String MATCH_ALL;
+   private static final String MATCH_ALL_WITH_SORT_BY_MESSAGE;
+   private static final String MATCH_ALL_WITH_SORT_BY_MSG;
+
+   static {
+       try {
+           MATCH_ALL = Files.readString(Paths.get("src/test/resources/common/matchAllQuery.json"));
+           MATCH_ALL_WITH_SORT_BY_MESSAGE = Files.readString(Paths.get(TEST_DIR, "matchAllWithSortByMessageQuery.json"));
+           MATCH_ALL_WITH_SORT_BY_MSG = Files.readString(Paths.get(TEST_DIR,"matchAllWithSortByMsgQueryWithoutSize.json"));
+       } catch (IOException e) {
+           throw new RuntimeException(e);
+       }
+   }
```

Comment:
Probably better to put these into a `@BeforeAll` method like they're loaded elsewhere, simply call the parent's `super.setUpBeforeClass()` to load common files too

Reply:
@ChrisSamo632 Just would like to clarify this. You want me to pull all of these up into `AbstractJsonQueryElasticsearchTest` `setUpBeforeClass` method and not just MATCH_ALL which seems to be already loaded by `AbstractJsonQueryElasticsearchTest`?

Reply:
Yes, exactly that - you're correct that `super` isn't a thing (my bad), but you should be able to refer to `AbstractJsonQueryElasticsearchTest. setUpBeforeClass()`